test/MinimalDelegation.execute.invariant.t.sol

```diff
@@ -0,0 +1,245 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.23;
+
+import {console2} from "forge-std/console2.sol";
+import {ERC20} from "openzeppelin-contracts/contracts/token/ERC20/ERC20.sol";
+import {EnumerableSetLib} from "solady/utils/EnumerableSetLib.sol";
+import {ERC20Mock} from "openzeppelin-contracts/contracts/mocks/token/ERC20Mock.sol";
+import {Test} from "forge-std/Test.sol";
+import {TokenHandler} from "./utils/TokenHandler.sol";
+import {ExecuteHandler} from "./utils/ExecuteHandler.sol";
+import {DelegationHandler} from "./utils/DelegationHandler.sol";
+import {TestKeyManager, TestKey} from "./utils/TestKeyManager.sol";
+import {IMinimalDelegation} from "../src/interfaces/IMinimalDelegation.sol";
+import {INonceManager} from "../src/interfaces/INonceManager.sol";
+import {IERC7821} from "../src/interfaces/IERC7821.sol";
+import {IKeyManagement} from "../src/interfaces/IKeyManagement.sol";
+import {Call, CallLib} from "../src/libraries/CallLib.sol";
+import {KeyType, Key, KeyLib} from "../src/libraries/KeyLib.sol";
+import {HandlerCall, CallUtils} from "./utils/CallUtils.sol";
+import {WrappedDataHash} from "../src/libraries/WrappedDataHash.sol";
+import {FunctionCallGenerator} from "./utils/FunctionCallGenerator.sol";
+import {Settings, SettingsLib} from "../src/libraries/SettingsLib.sol";
+import {SettingsBuilder} from "./utils/SettingsBuilder.sol";
+import {SignedCalls, SignedCallsLib} from "../src/libraries/SignedCallsLib.sol";
+
+// To avoid stack to deep
+struct SetupParams {
+    IMinimalDelegation _signerAccount;
+    TestKey[] _keys;
+    address[] _callers;
+    address _tokenA;
+    address _tokenB;
+}
+
+contract MinimalDelegationExecuteInvariantHandler is ExecuteHandler, FunctionCallGenerator {
+    using TestKeyManager for TestKey;
+    using EnumerableSetLib for EnumerableSetLib.Bytes32Set;
+    using KeyLib for Key;
+    using CallUtils for Call;
+    using CallUtils for Call[];
+    using CallUtils for HandlerCall;
+    using CallUtils for HandlerCall[];
+    using SignedCallsLib for SignedCalls;
+    using SettingsBuilder for Settings;
+
+    address[] public callers;
+    address public currentCaller;
+
+    TestKey[] public keys;
+    TestKey public currentSigningKey;
+
+    ERC20Mock public tokenA;
+    ERC20Mock public tokenB;
+
+    IMinimalDelegation public signerAccount;
+
+    constructor(SetupParams memory _params) FunctionCallGenerator(_params._tokenA, _params._tokenB) {
+        signerAccount = _params._signerAccount;
+        for (uint256 i = 0; i < _params._keys.length; i++) {
+            keys.push(_params._keys[i]);
+        }
+        callers = _params._callers;
+        tokenA = ERC20Mock(_params._tokenA);
```

Comment:
could use TokenHandler?

Reply:
I wanted to pass the tokens in by reference here, so we can change the state within the invariant handler and test for it outside in the test file